### PR TITLE
refactoring to use index dataset

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
# People index

The people index (`people.ttl`) is now used to show, save and delete contacts. 

# Checklist

- [x] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by megoth — CHANGES_REQUESTED** [review]

File: components/contactsList/index.jsx [hunk 1]
```diff
@@ -138,7 +147,7 @@ function ContactsList() {
       profileIri={selectedContactWebId}
     />
   );
-
+  // return <div>Boo</div>;
```
[the remark this answers is not in the document]
Comment (VirginiaBalseiro) [reply components/contactsList/index.jsx:150]: looool but they're funny though

File: src/addressBook/index.test.js [hunk 2]
```diff
@@ -271,47 +272,73 @@ describe("getContacts", () => {
...
     jest
       .spyOn(resourceFns, "getResource")
-      .mockResolvedValueOnce({ response: { dataset: "people container" } })
       .mockResolvedValueOnce({ response: expectedPerson1 })
       .mockResolvedValueOnce({ response: expectedPerson2 });
 
     jest
-      .spyOn(solidClientFns, "getUrlAll")
-      .mockReturnValueOnce([expectedPerson1.iri, expectedPerson2.iri]);
+      .spyOn(solidClientFns, "getThingAll")
```
Comment (megoth) [inline src/addressBook/index.test.js:298]: You don't need to mock `getThingAll`, but make sure to provide the dataset you need it to process.

To test the filter function, maybe insert a thing that isn't a `foaf:Person` as well.

File: src/addressBook/index.test.js [hunk 3]
```diff
@@ -271,47 +272,73 @@ describe("getContacts", () => {
...
+    };
+    const mockIndexFileDatasetIri =
+      "https://user.example.com/contacts/people.ttl";
+    const mockIndexFileDataset = solidClientFns.mockSolidDatasetFrom(
+      mockIndexFileDatasetIri
+    );
     const fetch = jest.fn();
 
+    jest
+      .spyOn(solidClientFns, "getThingAll")
```
Comment (megoth) [inline src/addressBook/index.test.js:327]: Ditto (do not mock getThingAll).

File: src/addressBook/index.test.js [hunk 4]
```diff
@@ -433,33 +466,34 @@ describe("saveContact", () => {
...
-      createSolidDataset(),
-      createThing({ name: "this" })
+    const contactDataset = mockPersonDatasetAlice();
+
+    const peopleIndexIri = `${addressBookIri}/people.ttl`;
+    const peopleIndexDataset = solidClientFns.mockSolidDatasetFrom(
+      peopleIndexIri
     );
 
+    jest.spyOn(solidClientFns, "getSourceUrl").mockReturnValue(addressBookIri);
```
Comment (megoth) [inline src/addressBook/index.test.js:479]: Don't mock `getSourceUrl`.

File: src/addressBook/index.test.js [hunk 5]
```diff
@@ -433,33 +466,34 @@ describe("saveContact", () => {
...
+
+    jest
+      .spyOn(addressBookFns, "createContact")
```
Comment (megoth) [inline src/addressBook/index.test.js:482]: `createContact` isn't an asynchronous function that connects to the server, so you shouldn't need to mock it.

File: src/addressBook/index.test.js [hunk 6]
```diff
@@ -527,14 +575,19 @@ describe("deleteContact", () => {
...
     jest
-      .spyOn(resourceFns, "getResource")
-      .mockResolvedValueOnce({ response: addressBook.people });
+      .spyOn(solidClientFns, "getSolidDataset")
+      .mockResolvedValueOnce(peopleIndexDataset);
 
     jest
       .spyOn(solidClientFns, "removeThing")
-      .mockReturnValueOnce(newAddressBook.people.dataset);
+      .mockReturnValueOnce(updatedPeopleIndexDataset);
```
Comment (megoth) [inline src/addressBook/index.test.js:590]: No need to mock `removeThing`.

File: src/hooks/useContacts/index.test.jsx [hunk 7]
```diff
@@ -66,49 +64,43 @@ describe("usePeople", () => {
 
     afterEach(() => {
       cache.clear();
+      jest.clearAllMocks();
+      cleanup();
```
Comment (megoth) [inline src/hooks/useContacts/index.test.jsx:68]: I think https://testing-library.com/docs/react-testing-library/api/#cleanup is added automatically? Sure you need it here?

**Review by megoth — CHANGES_REQUESTED** [review]

**Review by ajacksified — COMMENTED** [review]
Looks good after test cleanup.

**Review by megoth — APPROVED** [review]

**Pull request merged**